tests/integration/checkpoint.bats

```diff
@@ -315,3 +315,56 @@ function simple_cr() {
 	unlink "$tmp"
 	test -f ./work-dir/"$tmplog2" && unlink ./work-dir/"$tmplog2"
 }
+
+@test "checkpoint and restore with nested bind mounts" {
+	bind1=$(mktemp -d)
+	bind2=$(mktemp -d)
+	update_config '	  .mounts += [{
+					type: "bind",
+					source: "'"$bind1"'",
+					destination: "/test",
+					options: ["rw", "bind"]
+				},
+	                        {
+					type: "bind",
+					source: "'"$bind2"'",
+					destination: "/test/for/nested",
+					options: ["rw", "bind"]
+				}]'
+
+	# Before starting the container mount a tmpfs at $bind1
+	mount -t tmpfs tmpfs "$bind1"
+
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
+	[ "$status" -eq 0 ]
+
+	testcontainer test_busybox running
+
+	# checkpoint the running container
+	runc --criu "$CRIU" checkpoint --work-path ./work-dir test_busybox
+	grep -B 5 Error ./work-dir/dump.log || true
+	[ "$status" -eq 0 ]
+
+	# after checkpoint busybox is no longer running
+	testcontainer test_busybox checkpointed
+
+	# umount tmpfs at $bind1
+	umount "$bind1"
+	# mount it again so that runc re-creates missing mountpoints
+	mount -t tmpfs tmpfs "$bind1"
+
+	# restore from checkpoint
+	runc --criu "$CRIU" restore -d --work-path ./work-dir --console-socket "$CONSOLE_SOCKET" test_busybox
+	grep -B 5 Error ./work-dir/restore.log || true
+	[ "$status" -eq 0 ]
+
+	# busybox should be back up and running
+	testcontainer test_busybox running
+
+	# Clean up
```

Comment:
This is problematic -- if the test ever fails, it exits and this code won't be run, ending up with a dangling mount.

Looks like this is easy to solve by not using a tmpfs mount on the host, and instead of remounting $bind1 use `rm -rf $bind1/*`.

Reply:
Also the `teardown_*` commands aren't needed here because they're already present in the file-level `teardown` hook.

Reply:
I changed it as proposed by @kolyshkin. I initially repeated the code from the `teardown` because I wanted to make sure the directories are gone after the test. It would only have cleaned up after a successful test, but as that is the expected result I went for that.

I am now no longer using the tmpfs, but I am just deleting the created files before restore and I added a conditional cleanup block to existing file-level `teardown`.